src/SpendPermissionManager.sol

```diff
@@ -200,10 +203,10 @@ contract SpendPermissionManager is EIP712 {
     /// @param withdrawAmount Amount of asset attempting to withdraw from MagicSpend.
     error SpendValueWithdrawAmountMismatch(uint256 spendValue, uint256 withdrawAmount);
 
-    /// @notice Withdraw request nonce does not encode the correct spender address.
-    /// @param encoded The lower 160 bits of the nonce typed as an address.
-    /// @param expected Expected spender address.
-    error InvalidWithdrawRequestSpender(address encoded, address expected);
+    /// @notice Withdraw request nonce does not encode the correct spend permission hash
+    /// @param encoded The portion of the nonce that should match the spend permission hash
+    /// @param expected Expected partial hash of the spend permission
+    error InvalidWithdrawRequestHash(uint256 encoded, uint256 expected);
```

Comment:
thoughts on these renames?
```suggestion
    /// @notice Withdraw request nonce is not prefixed with the spend permission hash.
    /// @param permissionPrefix The first 16 bytes of the spend permission hash
    /// @param noncePrefix The first 16 bytes of the bytes32-cast of `WithdrawRequests.nonce`
    error InvalidWithdrawRequestNonce(bytes16 permissionPrefix, bytes16 noncePrefix);
```

Reply:
I like it! when casting like this we start dealing with lower bits (i.e. we grab lower bits of the hash and we place that in the lower bits of the nonce) v.s. upper bits that I was grabbing by right-shifting.  I think it's all arbitrary, and easier to understand by casting the way you've suggested, so I think lower bits are fine and I just adjusted the language to reflect that.